src/hypothesis/_settings.py

```diff
@@ -175,8 +176,28 @@ def defaults_stack(self):
             return self.storage.defaults_stack
 
     def __call__(self, test):
+        """Attach the settings as an attribute.
+
+        Also, so that we can issue a deprecation warning for
+        ``settings``` used alone, give note the deprecation, but also
+        attach the original test for ``@given`` to unwrap and avoid the
+        warning.
+        """
+        @proxies(test)
```

Comment:
Um.  This wasn't ever in scope, but now that we're here...

- What happens if you call a settings object *not* as a decorator?  (Nothing useful)
- What if someone applies `@settings(...)` multiple times, eg once for each option?  (only the last is kept - this is bad!)

So... well, it's easiest to explain my suggestion by writing it out:

```python
from hypothesis.internal.reflection import get_pretty_function_description

if not callable(test):
    raise InvalidArgument(
        'settings objects can be called as a decorator with @given, but '
        'test=%r' % (test,)
    )
if hasattr(test, '_hypothesis_internal_use_settings'):
    note_deprecation(
        '%s has already been decorated with a settings object, which will '
        'be overridden.  This will be an error in a future version of '
        'Hypothesis.\n    Previous:  %r\n    This:  %r' % (
            get_pretty_function_description(test),
            test._hypothesis_internal_use_settings,
            self
        )
    )
test._hypothesis_internal_use_settings = self
# This conditional avoids the warning when @settings is outer.
# (Unwrapping in @given avoids it when @settings is outer.)
if getattr(test, 'is_hypothesis_test', False):
    return test

@proxies(test)
def new_test(*args, **kwargs):
    note_deprecation(
        'Using `@settings` without `@given` does not make sense and '
        'will be an error in a future version of Hypothesis.'
    )
    test(*args, **kwargs)

new_test._hypothesis_internal_test_function_without_warning = test
new_test._hypothesis_internal_use_settings = self  # for double-use check
return new_test
```

You'll also need to write more tests if you go down this route, though the existing ones are perfect for what you've already done :smile: